pkg/docker-util/run.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerutil
+
+import (
+	"bytes"
+	"fmt"
+
+	dockerTypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	docker "github.com/docker/docker/client"
+
+	naglfarv1 "github.com/PingCAP-QE/Naglfar/api/v1"
+)
+
+type RunOptions struct {
+	Policy           naglfarv1.PullImagePolicy
+	Config           *container.Config
+	HostConfig       *container.HostConfig
+	NetworkingConfig *network.NetworkingConfig
+}
+
+func (client *Client) Run(name, image string, options RunOptions) (stat dockerTypes.ContainerJSON, requeue bool, err error) {
```

Comment:
`requeue` is an attribute of the kubebuilder framework, it's not very fit to use here IMO.

Reply:
resolved